plugins/out_es/es.c

```diff
@@ -597,8 +599,14 @@ void cb_es_flush(void *data, size_t bytes,
             ret = elasticsearch_error_check(c);
             if (ret == FLB_TRUE) {
                 /* we got an error */
-                flb_warn("[out_es] Elasticsearch error\n%s",
-                         c->resp.payload);
+                if (ctx->trace_error) {
+                    /*
+                     * If trace_error is set, trace the actual
+                     * input/output to Elasticsearch that caused the problem.
+                     */
+                    flb_error(stderr, "[out_es] error: Input\n%s\nOutput\n%s\n",
```

Comment:
flb_error() prototype don't need to specify the output stream since it's handled internally, also the extra \n is not required.